src/Microsoft.Health.Fhir.Core.UnitTests/Features/Operations/Export/ExportJobTaskTests.cs

```diff
@@ -1454,6 +1476,243 @@ public async Task GivenAGroupExportJobWithANonExistantGroup_WhenExecuted_ThenThe
+                anonymizationConfigurationLocation: "anonymization-config-file");
+
+            SetupExportJobRecordAndOperationDataStore(exportJobRecordWithOneResource);
+
+            // First search should not have continuation token in the list of query parameters.
+            _searchService.SearchAsync(
+                Arg.Any<string>(),
+                Arg.Any<IReadOnlyList<Tuple<string, string>>>(),
+                _cancellationToken)
+                .Returns(x =>
+                {
+                    return CreateSearchResult(new[]
+                        {
+                            new SearchResultEntry(
+                                new ResourceWrapper(
+                                    "1",
+                                    "1",
+                                    "Patient",
+                                    new RawResource("data", Core.Models.FhirResourceFormat.Json),
+                                    null,
+                                    DateTimeOffset.MinValue,
+                                    false,
+                                    null,
+                                    null,
+                                    null)),
+                        });
+                });
+
+            IAnonymizer anonymizer = Substitute.For<IAnonymizer>();
+            IAnonymizerFactory factory = Substitute.For<IAnonymizerFactory>();
+
+            anonymizer.Anonymize(Arg.Any<ResourceElement>()).Returns(
+                _ =>
+                {
+                    capturedSearch = true;
+                    return new ResourceElement(ElementNode.FromElement(ElementNode.ForPrimitive("anonymized-resource")));
+                });
+            factory.CreateAnonymizerAsync(Arg.Any<string>(), Arg.Any<CancellationToken>()).Returns(_ => Task.FromResult<IAnonymizer>(anonymizer));
+            var inMemoryDestinationClient = new InMemoryExportDestinationClient();
+
+            var anonymizedExportJobTask = new ExportJobTask(
+                () => _fhirOperationDataStore.CreateMockScope(),
+                Options.Create(_exportJobConfiguration),
+                () => _searchService.CreateMockScope(),
+                _groupMemberExtractor,
+                _resourceToByteArraySerializer,
+                inMemoryDestinationClient,
+                _resourceDeserializer,
+                factory.CreateMockScope(),
+                NullLogger<ExportJobTask>.Instance);
+
+            await anonymizedExportJobTask.ExecuteAsync(_exportJobRecord, _weakETag, _cancellationToken);
+
+            string exportedValue = inMemoryDestinationClient.GetExportedData(new Uri(PatientFileName, UriKind.Relative));
+
+            Assert.Equal("anonymized-resource", exportedValue);
+            Assert.True(capturedSearch);
+        }
+
+        [Fact]
+        public async Task GivenAnonymizationConfigurationNotExist_WhenExecuted_ThenJobStatusShouldBeUpdatedToFailed()
```

Comment:
Could some of these anonymizer exception tests be combined into a Theory with InlineData to give the exception type?

Reply:
Thanks, updated.